packages/nuxt/src/runtime/hooks/updateRouteBeforeResponse.ts

```diff
@@ -0,0 +1,49 @@
+import { getActiveSpan, getCurrentScope, getRootSpan, logger, SEMANTIC_ATTRIBUTE_SENTRY_SOURCE } from '@sentry/core';
+import type { H3Event } from 'h3';
+
+/**
+ * Update the root span (transaction) name for routes with parameters based on the matched route.
+ */
+export function updateRouteBeforeResponse(event: H3Event): void {
+  if (event.context.matchedRoute) {
+    const matchedRoutePath = event.context.matchedRoute.path;
+
+    // If the matched route path is defined and differs from the event's path, it indicates a parametrized route
+    // Example: Matched route is "/users/:id" and the event's path is "/users/123",
+    if (matchedRoutePath && matchedRoutePath !== event._path) {
+      if (matchedRoutePath === '/**') {
+        // todo: support parametrized SSR pageload spans
+        // If page is server-side rendered, the whole path gets transformed to `/**` (Example : `/users/123` becomes `/**` instead of `/users/:id`).
+        return; // Skip if the matched route is a catch-all route.
+      }
+
+      const method = event._method || 'GET';
+
+      const parametrizedTransactionName = `${method.toUpperCase()} ${matchedRoutePath}`;
+      getCurrentScope().setTransactionName(parametrizedTransactionName);
+
+      const activeSpan = getActiveSpan(); // In development mode, getActiveSpan() is always undefined
+      if (activeSpan) {
+        const rootSpan = getRootSpan(activeSpan);
+        if (rootSpan) {
+          rootSpan.updateName(parametrizedTransactionName);
+          rootSpan.setAttributes({
+            [SEMANTIC_ATTRIBUTE_SENTRY_SOURCE]: 'route',
+            'http.route': matchedRoutePath,
+          });
+
+          const params = event.context?.params || null;
```

Comment:
l: I would like us to avoid mixing `undefined` and `null`. Just doing `const params = event.context?.params` and having the `if (params)` should be sufficient.